sql/analyzer/filters.go

```diff
@@ -18,11 +18,15 @@ func (f filters) merge(f2 filters) {
 func exprToTableFilters(expr sql.Expression) filters {
 	filtersByTable := make(filters)
 	for _, expr := range splitExpression(expr) {
+		var seenTables = make(map[string]struct{})
 		var tables []string
```

Comment:
do we need `var tables []string` now? If we just check:
```
if len(tables) == 1 {
 			filtersByTable[tables[0]] = append(filtersByTable[tables[0]], expr)		 			filtersByTable[tables[0]] = append(filtersByTable[tables[0]], expr)
 		}		 		}
```
?

Reply:
How do you get the key from the map, then? You would need to range the map and that feels like this with extra steps

Reply:
that's true - we'll need to `range` and break, but on the other hand if we really need just 0th element then we may just memoize just one string instead of all slice and only set it `if len(seen) == 0`

Reply:
Fixed